Source/cmMakefileLibraryTargetGenerator.cxx

```diff
@@ -904,7 +909,8 @@ void cmMakefileLibraryTargetGenerator::WriteLibraryRules(
   if (useLinkScript) {
     // Use a link script.
     const char* name = (relink ? "relink.txt" : "link.txt");
-    this->CreateLinkScript(name, real_link_commands, commands1, depends);
+      this->CreateLinkScript(name, real_link_commands, commands1, depends);
+      this->CreateLinkScriptJSON(name, files, real_link_commands);
```

Comment:
What happens in case files are empty since useArchiveRules = false? There will be a command with empty files? Could you please provide an example for such case?

Reply:
Well, seems like I had a mistake in November. useArchiveRules and empty files are not an issue.

Reply:
But what if useLinkScript is false? Seems like cmake should build files, however the information about a command will not present in link_commands.json. Is that right? What is useLinkScript? In what cases it can be false? What happens if it is false?